### PR TITLE
Task #202906 Search filter by first name and last name added to the learner admin list 

### PR DESCRIPTION
**Pull request by GitVivekHub**
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

### CODE REVIEW

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 1]
```diff
@@ -502,6 +502,23 @@ export class BeneficiariesService {
 		filterQueryArray.push(
 			`{ program_beneficiaries: { facilitator_user: { program_faciltators: { parent_ip: { _eq: "${user?.data?.program_users[0]?.organisation_id}" } } } } }`,
 		);
+		if (body.search && body.search !== '') {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:505]: add blank line before if

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 2]
```diff
@@ -502,6 +502,23 @@ export class BeneficiariesService {
...
+				{ last_name: { _ilike: "%${last_name}%" } }
+				 ]} `);
+		 } else{
+            	
+			filterQueryArray.push(`{_or: [
+				{ first_name: { _ilike: "%${first_name}%" } }
+				 ]} `);
+		 }
+		}
 		if (body?.district && body?.district.length > 0) {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:522]: add blank line before if

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 2 above)
Comment (GitVivekHub) [reply src/src/beneficiaries/beneficiaries.service.ts]: resolved comment

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 1 above)
Comment (GitVivekHub) [reply src/src/beneficiaries/beneficiaries.service.ts]: resolved comment

**Pull request merged**